### ISSUE TITLE
lower-level v1.DatastoreClient class under the hood ??

### ISSUE THREAD

**Issue by trollkotze** [googleapis/nodejs-datastore#85]
This may not be an error, but it is a point of confusion.

What is the purpose of all the code within the subdirectory "v1"? Its main content seems to be the class "DatastoreClient", defined in the file named "datastore_client.js".

In the [docs](https://cloud.google.com/nodejs/docs/reference/datastore/1.4.x/Datastore) describing the `Datastore` constructor, we have:

>  Idiomatic class for interacting with Cloud Datastore. Uses the lower-level
>   [v1.DatastoreClient](https://cloud.google.com/nodejs/docs/reference/datastore/1.4.x/v1.DatastoreClient) class under the hood.

[v1.DatastoreClient](https://cloud.google.com/nodejs/docs/reference/datastore/1.4.x/v1.DatastoreClient) is listed in "classes".

The `v1.DatastoreClient` class by itself seems to provide similar or the same functionality as the main export `Datastore`, but at a lower level.

However, the statement that the main export `Datastore` "uses the lower-level `v1.DatastoreClient` class under the hood", is not true. The only bit used is the static class member `scopes`, which is defined as (a getter for) the following array:

```
  static get scopes() {
    return [
      'https://www.googleapis.com/auth/cloud-platform',
      'https://www.googleapis.com/auth/datastore',
    ];
```

I am scratching my head, asking myself what is the purpose of this lower-level `v1.DatastoreClient` class? What are the use cases where one might want to use it?

(I have actually just stumbled across it while scouring for hints on handling errors: What types of errors are possible to occur, and how should they be handled?
The specific error I came across today was a "deadline exceeded". It would be really good to have a comprehensive guide on what different error types mean and how to best respond to them - apart from the obvious: retrying until successful.)

#### Environment details

  - OS: unimportant
  - Node.js version: unimportant
  - npm version: unimportant
  - `@google-cloud/datastore` version: 1.4.0

#### Steps to reproduce

  1. Read the docs, looking for hints on what types of errors to expect and how to best handle them.
  2. See reference to an enigmatic "lower-level [`v1.DatastoreClient`](https://cloud.google.com/nodejs/docs/reference/datastore/1.4.x/v1.DatastoreClient) class", stated to be "used under the hood".
  3. ???
  4. PROFIT

**Comment by stephenplusplus** [comment]
We do actually use it! :) All requests that we make from the various hand-made API methods funnel through [this request function](https://github.com/googleapis/nodejs-datastore/blob/0c8237e8f67853a5b87276324d1ce37650885bae/src/request.js#L1141). That creates the "GAPIC" client (which is what the auto-generated, v1 directory is).

Many years ago, we had one, big auto-generated library that included the code for *all* of the API services. There was nothing idiomatic about it-- no care was taken to the conventions of the language. So, we set out to create hand-made APIs, one by one. Datastore was one of the first, but others quickly followed. Storage, Pub/Sub, BigQuery, etc.

We started to hear that some people actually liked the generated libraries, because for whatever reasons, their use case scooted around our goal of 99% common-case coverage. In order to support that last 1%, we expose the generated code, so that every possible dial that can be turned and option that can be configured is accessible.

Regarding the error handling discussion, we can keep that over here: https://github.com/googleapis/nodejs-datastore/issues/83
